routes/articles.js

```diff
@@ -1,61 +1,178 @@
+      .returning('*')
+      .then((result) => {
+        msg = 'Article succesfully added';
+        return res.redirect('/articles');
+      })
+      .catch((err) => {
+        res.status(500).send('err' + err);
+      });
   });
 
 routerArticles.route('/new').get((req, res) => {
-  res.render('templates/articles/new', articleDB.retrieveAll());
+  let resultObj = {};
+  if (msg !== '') {
+    resultObj.message = msg;
+    msg = '';
+  }
+  res.render('templates/articles/new', resultObj);
 });
 
 routerArticles
   .route('/:title')
   .get((req, res) => {
     const articleUrl = req.url.slice(1);
-    const findArticle = articleDB.getIndex(articleUrl);
-    if (findArticle === -1) {
-      return res.redirect('/articles');
-    }
-    // if last message on specific article was for a failed edit ...
-    // then remove the message when routing to "/articles/:urlTitle" ...
-    // since message was displayed originally on re-route to "/articles"
-    if (articleDB.getMessage(articleUrl) === '{ "success": false, "message": "No field values were provided" }') {
-      articleDB.removeMessage(articleUrl);
-    }
-    res.render(`templates/articles/article`, articleDB.retrieveOne(articleUrl));
-    articleDB.removeMessage(articleUrl);
+    knex
+      .select()
+      .table('articles')
+      .where('urlTitle', articleUrl)
+      .then((result) => {
+        if (result.length === 0) {
+          msg = 'Article not found';
+          return res.redirect('/articles');
+        }
+        let resultObj = result[0];
+        if (msg !== '') {
+          resultObj.message = msg;
+          msg = '';
+        }
+        res.render('templates/articles/article', resultObj);
+      })
+      .catch((err) => {
+        res.status(500).send(err);
+      });
   })
   .put((req, res) => {
     const endIndexQmark = req.url.indexOf('?', 1) - 1;
     const articleUrl = req.url.substr(1, endIndexQmark);
-    articleDB.update(req.body, articleUrl);
-    const errorMsg = articleDB.getMessage(articleUrl).indexOf('"success": false') > -1;
-    if (errorMsg) {
-      res.redirect('/articles');
-    } else {
-      res.redirect(`/article/${articleUrl}`);
-    }
+    knex
+      .select()
+      .table('articles')
+      .where('urlTitle', articleUrl)
+      .then((result) => {
+        if (result.length === 0) {
+          msg = 'Article not found';
+          return res.redirect('/articles');
+        }
+        let updateObj = {};
+        if (req.body.title) {
+          updateObj.title = req.body.title;
+          updateObj.urlTitle = encodeURI(req.body.title);
+        }
+        if (req.body.author) {
+          updateObj.author = req.body.author;
+        }
+        if (req.body.body) {
+          updateObj.body = req.body.body;
+        }
+
+        if (Object.keys(updateObj).length === 0) {
+          msg = 'No field values were provided';
+          return res.redirect(`/articles/${articleUrl}/edit`);
+        }
+        knex('articles')
```

Comment:
Return this promise then you let the catch in the upper promise handle the res.status 500